accord-core/src/main/java/accord/coordinate/tracking/ReadTracker.java

```diff
@@ -148,19 +228,26 @@ public Set<Id> computeMinimalReadSetAndMarkInflight()
         while (!toRead.isEmpty())
         {
             if (candidates.isEmpty())
+            {
+                if (!nodes.isEmpty())
+                    nodes.forEach(this::recordContacted);
```

Comment:
The premise here is that we _couldn't_ do anything, so these nodes are not useful anymore, and we want to avoid continually trying to contact someone for more information on one shard, when another cannot be met by any remaining candidates.

However, this is actually a little bit broken. I'm going to refactor it a little and introduce some dedicated tests for read tracking.